sim/warlock/pets.go

```diff
@@ -213,7 +213,7 @@ func (pet *WarlockPet) registerShadowBiteSpell() {
 
 			activeDots := 0
 			for _, spell := range pet.Owner.Spellbook {
-				if spell.Dot(target) != nil && spell.Dot(target).IsActive() {
+				if spell.ClassSpellMask&WarlockDoT > 0 && spell.Dot(target).IsActive() {
```

Comment:
That won't work, shadow bite scales with all dots, even ones from trinkets and such.

Reply:
I was just about to ask you if it should scale as I saw a test with a dot trinket was still failing.

Reply:
I can do something else which is a bit ugly but should work as expected:
```
if spell.Dot(target) != nil && spell.RelatedDotSpell == nil && spell.Dot(target).IsActive() {
	activeDots++
}
```

This would make it so spells that have a related dot spell are not counted as their related dot spell will be counted anyway

Reply:
Guess that works, gotta add a comment there though, no way someone reading this code understands why it's there 😂